package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -540,7 +597,12 @@ def _get_dh_pairs(self):
         # Otherwise, use d_h_cutoff as a cutoff distance
         else:
 
-            hydrogens = self.u.select_atoms(self.hydrogens_sel)
+            if self.hydrogens_sel is None:
```

Comment:
This doesn't change the default behavior since this method is called in `_prepare` and if `hydrogens_sel==None` then `self.guess_hydrogens()` is called first anyway. Because this is a hidden method, the only place this change makes a difference is in testing, which I believe is why I did it. Additionally, I think this change better aligns with the purpose of the function. The description of this method `_get_dh_pairs` is:

> Finds donor-hydrogen pairs.
>
> Returns
> donors, hydrogens: AtomGroup, AtomGroup
>&nbsp;&nbsp;&nbsp;&nbsp;AtomGroups corresponding to all donors and all hydrogens. AtomGroups are ordered such that, if zipped, will produce a list of donor-hydrogen pairs.

Reply:
As an aside, I would advocate that all instances of `variable is None` be changed to `variable == None` since all values of `None` don't necessarily come from the same memory location... I've been burned before.

Reply:
I was/am pretty sure None was a singleton in Python and always "is" itself.

Reply:
> As an aside, I would advocate that all instances of variable is None be changed to variable == None since all values of None don't necessarily come from the same memory location... I've been burned before.

So the main reason why we use `is None` over `== None` is due to a PEP8.

As far as I understand it, although `==` and `is` are fundamentally different, due to the nature of `None`, the two end up being equivalent? At least I don't know of an edge case where the comparison does not hold up. There is this specific case I believe, but I take it as a none-issue [pun intended ;) ] http://jaredgrubb.blogspot.com/2009/04/python-is-none-vs-none.html

Reply:
In the development of my own package I remember having versions of `None` with different memory locations. Unfortunately, I can't find my notes/emails on discussions of this topic and I can't reproduce it, so I will yield.

Reply:
If you do find your notes on this I would be quite interested / open to further changes in follow-up work. The behaviour of `None` is quite odd compared to other comparisons, so it wouldn't surprise me if there were edge cases that haven't been considered.